File: Lab3OOP/InstrumentalWork.cpp
#include "InstrumentalWork.h"
#include <iostream>
#include <sstream>

InstrumentalWork::InstrumentalWork(const std::string& title, int duration, const std::string& composer) : Composition(title, duration), composer(composer) {}

std::string InstrumentalWork::show() const {
    std::ostringstream oss;
    oss << "Title: " << title << ", Duration: " << duration << " sec, Composer: " << composer;
    return oss.str();
}

std::string InstrumentalWork::getComposer() const {
    return composer;
}

File: Lab3_oop_python/instrumental_work.py
from composition import Composition

class InstrumentalWork(Composition):
    def __init__(self, title: str, duration: int, composer: str):
        super().__init__(title, duration)
        self.composer = composer

    def get_composer(self) -> str:
        return self.composer
    def show(self):
        print(f"Title: {self.title}, Duration: {self.duration} sec, Composer: {self.composer}")
